pom.xml

```diff
@@ -206,6 +206,12 @@ The following provides more details on the included cryptographic software:
         <target.name>linux-aarch64</target.name>
       </properties>
     </profile>
+    <profile>
+      <id>linux-riscv64</id>
+      <properties>
+        <target.name>linux-riscv64</target.name>
+      </properties>
+    </profile>
```

Comment:
To be 100% clear, it's not because we reference it here that JNA version 5.12.0 will be used either as a dependency to commons-crypto on linux-riscv64 only (given a single package is built and distributed, based on the linux-x86_64 build), either on CI or on anyone else's machine and deployment. Even if we add it here, and commons-crypto ships now with a dependency on 5.13.0, commons-crypto on linux-riscv64 will _not_ work out-of-the-box. We are doing all these 5.12.0 downgrade for CI purposes only. Luckily, that issue affects linux-riscv64 only and no other platform.

There are two possible fixes:
1. the proper one: JNA ships version 5.13.1 or 5.14.0 with the linux-riscv64 not depending on GLIBC 2.34. __And__ everyone depending on JNA (directly or transitively) upgrading to JNA 5.14.0+.
2. the workaround: anyone depending on commons-crypto manually cuts the commons-crypto -> JNA dependency and manually references JNA version 5.12.0 in their won pom.xml.

I expect the proper fix to be acceptable as things tend to converge long term, and anyone depending on JNA directly or transitively can easily override the version of JNA in their own pom.xml.

So once JNA 5.13.1 or 5.14.0 has shipped, we should remove that 5.12.0 downgrade for CI purposes, and to make sure commons-crypto keeps track of upstream JNA.

Reply:
To demonstrate that, an example to highlight the difference.

1. You build normally on a linux-x86_64 machine, and referencing JNA version 5.13.0
```
$> docker compose -f src/docker/docker-compose.yaml run crypto src/docker/build.sh
```

2. If you try running the tests normally, it will fail:
```
$> docker compose -f src/docker/docker-compose.yaml run crypto-riscv64 mvn -V -B -ntp surefire:test
[...]
[ERROR] org.apache.commons.crypto.jna.CbcNoPaddingCipherJnaStreamTest.testReadWrite -- Time elapsed: 0.022 s <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.apache.commons.crypto.jna.OpenSslNativeJna
        at org.apache.commons.crypto.jna.OpenSslJna.isEnabled(OpenSslJna.java:89)
        at org.apache.commons.crypto.jna.AbstractCipherJnaStreamTest.init(AbstractCipherJnaStreamTest.java:37)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.UnsatisfiedLinkError: /root/.cache/JNA/temp/jna6069377359221932581.tmp: /lib/riscv64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /root/.cache/JNA/temp/jna6069377359221932581.tmp) [in thread "main"]
        at java.base/jdk.internal.loader.NativeLibraries.load(Native Method)
        at java.base/jdk.internal.loader.NativeLibraries$NativeLibraryImpl.open(NativeLibraries.java:331)
        at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:197)
        at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:139)
        at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2418)
        at java.base/java.lang.Runtime.load0(Runtime.java:852)
        at java.base/java.lang.System.load(System.java:2021)
        at com.sun.jna.Native.loadNativeDispatchLibraryFromClasspath(Native.java:1045)
        at com.sun.jna.Native.loadNativeDispatchLibrary(Native.java:1015)
        at com.sun.jna.Native.<clinit>(Native.java:221)
        at com.sun.jna.NativeLibrary.<clinit>(NativeLibrary.java:90)
        at org.apache.commons.crypto.jna.OpenSslNativeJna.<clinit>(OpenSslNativeJna.java:59)
        ... 5 more
```

3. But if you override the JNA version (even though it doesn't rebuild! it's only a run-time dependency), it succeeds:
```
$> docker compose -f src/docker/docker-compose.yaml run crypto-riscv64 mvn -V -B -ntp surefire:test -Djna.version=5.12.0
```

`surefire:test` doesn't rebuild anything, and only makes sure that the right packages are on the CLASSPATH. 

That can be reproduced by any user of commons-crypto by specifying a newer version of JNA directly. That would obviously be useful _only_ for the linux-riscv64 case, every other platforms "just works" because they don't have that issue.